parts/linux/cloud-init/artifacts/cse_helpers.sh

```diff
@@ -1040,4 +1042,60 @@ extract_tarball() {
     esac
 }
 
-#HELPERSEOF
+handleCloudInitStatus() {
+    local provisionOutput="$1"
+    local cloudInitMessage=""
+    
+    # Capture detailed cloud-init status for tracking of different errors
+    cloud-init status --wait > /dev/null 2>&1;
+    local cloudInitExitCode=$?;
+
+    if [ "$cloudInitExitCode" -eq 0 ]; then
+        cloudInitMessage="cloud-init succeeded"
+        echo "${cloudInitMessage}" >> "${provisionOutput}";
+        # we want to avoid logging more information for success case (majority of them)
+        return 0
+    elif [ "$cloudInitExitCode" -eq 1 ]; then
```

Comment:
this statement for when the exit code is 1 is actually redundant right? since we would have exited earlier.

Reply:
Yeah, I left it like this in case something changes in `cse_cmd.sh`, just in case and to make sure `handleCloudInitStatus` covers all the cases